src/Share/Postgres/NameLookups/Queries.hs

```diff
@@ -501,6 +502,37 @@ projectTypesWithinRoot !_nlReceipt bhId = do
     |]
     <&> fmap (\NamedRef {reversedSegments, ref} -> (reversedNameToName reversedSegments, ref))
 
+-- | Get a cursor over all non-lib, non-builtin types and their corresponding constructor counts, within the given root branch.
+projectConstructorCountsWithinRoot :: (QueryM m) => NameLookupReceipt -> BranchHashId -> m (PGCursor (TypeReferenceId :. Only Int64))
+projectConstructorCountsWithinRoot !_ bhId =
```

Comment:
Do you want the number of constructors which also have names, or is it good enough to get the number of constructors that a given type has? 

This appears to only count constructors which have names in the current namespace, not sure if that's what you want here or not :)